tensorflow/core/kernels/cudnn_rnn_ops.cc

```diff
@@ -650,20 +664,29 @@ Status DoForward(OpKernelContext* context, const RnnDescriptor& rnn_desc,
                  Tensor* output, Tensor* output_h, Tensor* output_c,
                  ScratchAllocator* reserve_space_allocator,
                  ScratchAllocator* workspace_allocator,
-                 ProfileResult* output_profile_result) {
+                 ProfileResult* output_profile_result,
+                 const Tensor* sequence_lengths = nullptr) {
   std::unique_ptr<RnnSequenceTensorDescriptor> input_desc;
   std::unique_ptr<RnnStateTensorDescriptor> state_desc;
   std::unique_ptr<RnnSequenceTensorDescriptor> output_desc;
 
+  DeviceMemory<int> sequence_lengths_data;
+  int* seq_lens = nullptr;
+  if (sequence_lengths != nullptr) {
```

Comment:
How about trying `absl::Span`? There's one `MakeSpan(T* ptr, size_t size)`
https://github.com/abseil/abseil-cpp/blob/master/absl/types/span.h#L176

Let's pass a `absl::Span` to the sequence descriptor constructor?